_docs/runtime/installation.md

```diff
@@ -1,10 +1,91 @@
+  Install the Codefresh CLI using the option that best suits you: Curl, Brew, or standard download. If you are not sure which OS to select for Curl and Brew, simply select one, and Codefresh automatically identifies and selects the right OS for CLI installation.
+
+
+  
+### Installing the Codefresh runtime
+
+#### Runtime components
+ CSDP  create will create a repo (if not exist) and bootstrap (if needed) a new installation, levering the argocd-autopilot. The CLI will determine whether the new runtime is an overlay of an existing runtime or a completely new runtime.
+
+ 
+**Argo CD components**
+
+* Two Github repositories, one for the runtime’s infrastructure, and one for the first GitOps source defined through the default Git Source entity
+* Project, comprising an Argo CD AppProject and an ApplicationSet
+* Installations of the following applications in the project:
+  * Argo CD 
+  * Argo Workflows 
+  * Argo Events
+  * Argo Rollouts
+  
+**Codefresh-specific components**
+* Codefresh Applications in the Argo CD AppProject:  
+  * ArgoCD Agent that feeds GitOps data into CSDP
+  * App-proxy facilitating behind-firewall access to Git 
+  * Git Source entity that references a specific directory on a different repository
+*  A Service that pulls the changes from the repository defined in the Git Source, analyses them, and generates events for the created/updated/deleted files to CSDP
+*  Event Sources and Sensors for the K8s cluster, Git Sources, and CSDP 
+   (NIMA: should we detail as below?
+  * An EventSource and Sensor that sets a WebHook on the Git source, and triggers the above service
+  *  An EventSource and Sensor that registers to events on the K8s cluster, and reports them to CSDP
+  *  Components-reporter, an EventSource and Sensor that listens to Codefresh components and reports events to CSDP)
+
+#### Runtime prerequisites
+Before you install a runtime, make sure you have the following: 
+
+* **Ingress controller in Kubernetes cluster** 
+  Configure your Kubernetes cluster with an Ingress controller component, that is exposed from the cluster. (NIMA: Should we add what CF supports? Do we have limitations?) For detailed information, see the [Kubernetes documentation on ingress controllers](https://kubernetes.io/docs/concepts/services-networking/ingress-controllers/).
+  Note down the external address of the Ingress controller, as you will need this when you install the runtime. 
+
+* **Default `IngressClass`** 
+  If you do not have any `IngressClass` resource tagged as the default, add this `ingressclass.kubernetes.io/is-default-class` annotation with the value of `true` to the resource.
+  
+  ```yaml
+  apiVersion: networking.k8s.io/v1
+  kind: IngressClass
+  metadata:
+  annotations:
+    ingressclass.kubernetes.io/is-default-class: "true" 
+  ```
+
+* **Git tokens**  
+  You need a personal access Git token for authentication to the Git installation repo, that you will create or select during runtime installation.   
```

Comment:
we should explain differnt git proivders here.
this is for github, we should already prepare to split by provider

Reply:
Will update as needed